ndbench-core/src/main/java/com/netflix/ndbench/core/util/ConstantStepWiseRateIncreaser.java

```diff
@@ -0,0 +1,132 @@
+     */
+    public ConstantStepWiseRateIncreaser(int rampPeriodMillisecs,
+                                  int incrementIntervalMillisecs,
+                                  int initRate,
+                                  int finalRate) {
+        if (!(initRate >= 0)) {
+            throw new IllegalArgumentException("initRate must be >= 0");
+        }
+        if (!(finalRate > 0)) {
+            throw new IllegalArgumentException("finalRate must be > 0");
+        }
+        if (!(finalRate > initRate)) {
+            throw new IllegalArgumentException("finalRate must be > initRate");
+        }
+        if (!(rampPeriodMillisecs > 0)) {
+            throw new IllegalArgumentException("rampPeriodMillisecs must be > 0");
+        }
+        if (!(incrementIntervalMillisecs > 0)) {
+            throw new IllegalArgumentException("incrementIntervalMillisecs must be > 0");
+        }
+        if (rampPeriodMillisecs % incrementIntervalMillisecs != 0) {
+            throw new IllegalArgumentException(
+                    "rampPeriodMillisecs should be evenly divisible by incrementIntervalMillisecs");
+        }
+        if (rampPeriodMillisecs / incrementIntervalMillisecs > MAX_STEPS) {
+            throw new IllegalArgumentException(
+                    "rampPeriodMillisecs / incrementIntervalMillisecs should not exceed MAX_STEPS (" + MAX_STEPS + ")");
+        }
+
+
+        final ArrayList<Cell> tmpClockTimeToRateTable = new ArrayList<Cell>();
+
+        int numSteps = rampPeriodMillisecs / incrementIntervalMillisecs;
+        double spread = (finalRate - initRate) * 1.0;
+        double rateIncrementPerStep = spread / numSteps;
+
+        for (int i = 0; i <= numSteps; i++) {
+            tmpClockTimeToRateTable.add(
+                    new Cell(
+                            i * incrementIntervalMillisecs,
+                            initRate + (i * rateIncrementPerStep)
+                    )
+            );
+        }
+
+        // initialize high values sentinel cell
+        tmpClockTimeToRateTable.add(new Cell(Long.MAX_VALUE, 0 /* dummy never used for rate */));
+
+        indexOfLastInTable = numSteps + 1;
+        assert (tmpClockTimeToRateTable.size() == indexOfLastInTable + 1);
+        assert (tmpClockTimeToRateTable.get(numSteps).time == rampPeriodMillisecs);
+        assert (tmpClockTimeToRateTable.get(numSteps).rate == finalRate);
+
+
+        ImmutableList.Builder<Cell> cellBuilder = new ImmutableList.Builder<>();
+        clockTimeToRateTable = cellBuilder.addAll(tmpClockTimeToRateTable).build();
+    }
+
+    /**
+     * Build internal array mapping relative-from-zero clock time values to the rate that will be returned once system
+     * clock exceeds time value in a particular cell (but is less than the time value in the immediately
+     * succeeding cell);
+     */
+
+    public double getRateForGivenClockTime(long baseReferenceTime, long clockTime) {
```

Comment:
Can this function be replaced by the following formula: `initRate + (clockTime / incrementIntervalMillisecs) * rateIncrementPerStep`? This way you won't need to build and maintain that `clockTimeToRateTable` lookup table.